firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionLifecycleService.kt

```diff
@@ -142,7 +144,37 @@ internal class SessionLifecycleService() : Service() {
     }
 
     /** Generates a new session id and sends it everywhere it's needed */
-    private fun newSession() {
+    private suspend fun newSession() {
+      val subscribers = FirebaseSessionsDependencies.getRegisteredSubscribers()
+
+      if (subscribers.isEmpty()) {
+        Log.d(
+          TAG,
+          "Sessions SDK did not have any dependent SDKs register as dependencies. Events will not be sent."
+        )
+        return
+      }
+
+      if (subscribers.values.none { it.isDataCollectionEnabled }) {
+        Log.d(TAG, "Data Collection is disabled for all subscribers. Skipping this Session Event")
+        return
+      }
```

Comment:
Why do anything with subscribers from the service? It's the client's responsibility to interact with the subscribers on their containing processes.

Reply:
Done